apps/api/src/lesson/adminLesson.service.ts

```diff
@@ -114,6 +130,166 @@ export class AdminLessonService {
     });
   }
 
+  async createQuizLessonWithQuestionsAndOptions(
+    data: CreateQuizLessonBody,
+    authorId: UUIDType,
+    displayOrder: number,
+  ) {
+    return await this.db.transaction(async (trx) => {
+      const [lesson] = await trx
```

Comment:
DB calls are doing in repository, here we keep results and execute next repository functions